src/glmfit.jl

```diff
@@ -233,10 +233,13 @@ function _fit!(m::AbstractGLM, verbose::Bool, maxIter::Integer, minStepFac::Real
         installbeta!(p, f)
         crit = (devold - dev)/dev
         verbose && println("$i: $dev, $crit")
-        if crit < convTol
+        if crit < convTol || dev == 0
             cvg = true
             break
         end
+        if !isfinite(crit)
+            error("convergence criterion is $crit") # maybe change ConvergenceException to allow for custom error
+        end
```

Comment:
Or we could just change the internal field to a string and let people write their own messages, akin to base exception types.